File: reproduction/sample.py
class Calculator:
    def unused_method(self):
        """Method with no references."""
        return "never called"
    
    def add(self, a, b):
        """Method with 2 references."""
        return a + b
    
    def multiply(self, a, b):
        """Method with 3 references.""" 
        return a * b


# Create references
calc = Calculator()

# 2 references to add()
result1 = calc.add(1, 2)
result2 = calc.add(3, 4)

# 3 references to multiply()
result3 = calc.multiply(2, 3)
result4 = calc.multiply(4, 5)
result5 = calc.multiply(6, 7)

File: repro/sample.lua
-- Sample Lua file for lensline bug reproduction
-- This file contains functions with different reference counts for testing

-- Function with no references (0 refs)
local function unused_function()
    return "This function is never called"
end

-- Function with some references (2 refs)
local function helper_function(x, y)
    return x + y
end

-- Function with multiple references (3 refs)
local function popular_function(data)
    if not data then
        return nil
    end
    return data.value or "default"
end

-- Usage to create references
local result1 = helper_function(1, 2)
local result2 = helper_function(3, 4)

local value1 = popular_function({value = "test"})
local value2 = popular_function(nil)
local value3 = popular_function({other = "field"})

-- Main function
local function main()
    print("Sample Lua file loaded")
    print("Result:", result1, result2)
    print("Values:", value1, value2, value3)
end

return {
    main = main,
    helper = helper_function,
    popular = popular_function
}